minpy/array.py

```diff
@@ -422,6 +422,16 @@ def node(self):
         """ get node which contains derivative information from this array """
         return self._node
 
+    @property
+    def ndim(self):
+        """ Number of array dimensions """
+        if ArrayType.NUMPY in self._data:
+            return self._data[ArrayType.NUMPY].ndim
+        else:
+            # TODO add ndim in MXNet ndarray
+            # return self._data[ArrayType.MXNET].shape
+            return numpy.array(self.get_data(ArrayType.NUMPY)).ndim
```

Comment:
Please use `len(self.get_data(ArrayType.NUMPY).shape)` for current walkaround. Your code will incur a deep copy which is not needed.